helm/files/bootstrap.py

```diff
@@ -0,0 +1,354 @@
+        print("Network (Service Group)")
+        print("  ID:   {}".format(network["id"]))
+        print("  Name: {}\n".format(network["name"]))
+        configmap_data["network_id"] = str(network["id"])
+        configmap_data["network_name"] = network["name"]
+
+        dns_pool = mgr.create_dns_pool(operator, network)
+        if not dns_pool:
+            fail("could not create dns pool")
+
+        print("DNS Pool (Service Def)")
+        print("  ID:   {}".format(dns_pool["id"]))
+        print("  Name: {}\n".format(dns_pool["name"]))
+        configmap_data["service_def_id"] = str(dns_pool["id"])
+        configmap_data["service_def_name"] = dns_pool["name"]
+
+        org = mgr.create_org(operator, network)
+        if not org:
+            fail("could not create organization")
+
+        print("Organization")
+        print("  ID:   {}".format(org["id"]))
+        print("  Name: {}\n".format(org["name"]))
+        configmap_data["organization_id"] = str(org["id"])
+        configmap_data["organization_name"] = org["name"]
+
+        apikey = mgr.create_org_apikey(operator, org)
+        if not apikey:
+            fail("could not create api key")
+
+        print("ApiKey (Org)")
+        print("  Key: {}\n".format(apikey["key"]))
+        configmap_data["api_key_id"] = apikey["key"]
+
+        user = mgr.create_org_user(operator, org, name=FIRST_USER)
+        if not user:
+            fail("could not create user")
+
+        print("Portal User (Org)")
+        print("  Name: {}".format(user["username"]))
+        print("  Password: {}\n".format(user["password"]))
+        configmap_data["portal_user_username"] = user["username"]
+        configmap_data["portal_user_password"] = user["password"]
+
+        for i in range(1, 10):
+            mgr.create_scopegroup(
+                operator, "example_scope-{}".format(i), network, org
+            )
+
+        service = mgr.create_service(operator, "dhcpdemo1", network, org)
+        service = mgr.create_service(operator, "dhcpdemo2", network, org)
+
+        kubernetes.config.load_incluster_config()
+        k8s = kubernetes.client.CoreV1Api()
+
+        configmap_meta = kubernetes.client.V1ObjectMeta(
+            name="ns1-bootstrap-credentials"
+        )
+        configmap_body = kubernetes.client.V1ConfigMap(
+            data=configmap_data, metadata=configmap_meta
+        )
+
+        try:
+            k8s_response = k8s.replace_namespaced_config_map(
+                "ns1-bootstrap-credentials",
+                NAMESPACE,
+                configmap_body,
+                pretty="true",
+                field_manager="ns1-bootstrap",
+            )
+            pprint(k8s_response)
+        except ApiException as e:
+            print(
+                "Exception when calling CoreV1Api->replace_namespaced_config_map: %s\n"
+                % e
+            )
+
+    except BootstrapError as e:
+        fail("Exception catched: {}".format(e))
```

Comment:
I don't think this rather large try-except block is that useful.  Generally, one tries to keep the number of statements in the try as small as possible in order to differentiate the cause of the error.  In this case, I'm not sure I see the value in catching any exception just to print it.